protocols/v2/noise-sv2/src/initiator.rs

```diff
@@ -151,26 +152,32 @@ impl Initiator {
     ///
     ///
     ///
-    pub fn step_2(&mut self, message: [u8; 170]) -> Result<NoiseCodec, Error> {
+    pub fn step_2(&mut self, message: [u8; 202]) -> Result<NoiseCodec, Error> {
+
+        let mut  elliswift_serialized: [u8; 64] = [0; 64];    
+        elliswift_serialized.clone_from_slice(&message[0..64]);
+        let remote_pub_key = &PublicKey::from_ellswift(ElligatorSwift::from_array(elliswift_serialized)).x_only_public_key().0.serialize();
         // 2. interprets first 32 bytes as `re.public_key`
         // 3. calls `MixHash(re.public_key)`
-        let remote_pub_key = &message[0..32];
+        //let remote_pub_key = &message[0..32];
         self.mix_hash(remote_pub_key);
 
         // 4. calls `MixKey(ECDH(e.private_key, re.public_key))`
         let e_private_key = self.e.secret_bytes();
         self.mix_key(&Self::ecdh(&e_private_key[..], remote_pub_key)[..]);
 
         // 5. decrypts next 48 bytes with `DecryptAndHash()` and stores the results as `rs.public_key` which is **server's static public key** (note that 32 bytes is the public key and 16 bytes is MAC)
-        let mut to_decrypt = message[32..80].to_vec();
+        // added additional 32 bytes because of the switch to elligatorswift encoding, wich is 64
+        // bytes instead of 32
+        let mut to_decrypt = message[64..80+32].to_vec();
```

Comment:
Similar as above, it's more readable to use constants that explain where the numbers come from.

Reply:
you are correct